### ISSUE TITLE
writeFile and readFile were removed (but remain in documentation)

### ISSUE THREAD

**Issue by jkarni** [haskell-streaming/streaming#40]
E.g., [here](https://github.com/haskell-streaming/streaming/blob/master/src/Streaming/Prelude.hs#L2484). I'm not sure if the intended behavior was actually to remove them or not, and what the proposed equivalent would be, but if told, am happy to submit a PR.

**Comment by andrewthad** [comment]
Removing them was intentional, but leaving references to them in the documentation was not. The appropriate equivalent is a combination of `withFile` (from `System.IO`) and `fromHandle`.

**Comment by tomjaguarpaw** [comment]
```haskell
writeFile :: FilePath -> S.Stream (S.Of String) IO r -> IO r
writeFile f = System.IO.withFile f System.IO.WriteMode . flip S.toHandle
```

Here's the implementation for writing.  I'm not completely sure why this wouldn't be included in `streaming` as a utility function.

**Comment by tomjaguarpaw** [comment]
@andrewthad could you provide some context about why these were removed?  Here's the relevant commit

https://github.com/haskell-streaming/streaming/commit/ad28c447d0f10150a27e5cf08c322bdcbde39577#diff-ba6fdfabef0ae755277dddeacfc79bb8

**Comment by chessai** [comment]
@tomjaguarpaw these were removed because they can leak file descriptors. Something like:

```haskell
S.take $ S.readFile "file.txt"
```

could cause the file handle to remain open.

**Comment by tomjaguarpaw** [comment]
Could the versions in terms of `withFile` be added then?

**Comment by andrewthad** [comment]
@tomjaguarpaw Do you mean something like this:

    readFile :: String -> (Stream (Of String) IO r -> IO a) -> IO a
    writeFile :: String -> Stream (Of String) IO r -> IO r

**Comment by andrewthad** [comment]
If so, yes. I think those are good definitions.

**Comment by chessai** [comment]
@andrewthad i think the type signature is wrong on readFile, it should be

```haskell
readFile :: String -> (Stream (Of String) IO () -> IO a) -> IO a
```

**Comment by andrewthad** [comment]
Yeah, that's what it is supposed to be.

**Comment by chessai** [comment]
Can this be closed due to #55?

**Comment by chessai** [comment]
Closed due to #55
